### PR TITLE
Tighten design constraints

### PR DESCRIPTION
**Pull request by nickdrozd**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

This PR tightens a few of the design constraints in the config file. No code is changed, although a few disable comments are added.

The rationale for the numbers is that they should be as low as possible without raising too many warnings. This way only the worst offenders are flagged. For example, there are only a few functions with more than 50 statements, and more with between 40 and 50 statements. So 50 is the cutoff. I'm not wed to any of the numbers in particular, but I think they're about where they should be.

Observe that several functions raise multiple warnings. Such functions are prime candidates for refactoring.

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -413,16 +413,16 @@ max-spelling-suggestions=2
 max-args=10
 
 # Maximum number of locals for function / method body
```
Comment (Pierre-Sassoulas) [inline pylintrc:415]: Could we use the default value and remove the description and the value ?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you ! Let's cleanup the pylintrc and let's merge.

File: pylintrc (hunk 1 above)
Comment (nickdrozd) [reply pylintrc:415]: You mean for all of them or for max locals in particular? For all of the settings in this PR at least, the configured value is higher than the default, so using the defaults would require a lot more disabling comments.

As for the rest of the config file, personally I would be happy to rip through and cut out a bunch of the text. Currently there's an awful lot of text, and I find it hard to read. How much cleanup were you thinking?

Comment (Pierre-Sassoulas) [reply pylintrc:415]: I was thinking of removing the default values and their description. It should be possible to do that automatically with the conf génération command iirc

Comment (nickdrozd) [reply pylintrc:415]: I'm not quite sure what you mean.

Comment (nickdrozd) [reply pylintrc:416]: Okay, I cut a few more of them.

**Comment by nickdrozd** [comment]
Unrelated to this PR, it looks like the full set of primer tests runs even if linting fails. Is that something that should be changed? It seems wasteful to rerun everything after lint fixups.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you nice clean up !

**Pull request merged**